jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/AndroidVariantEndpoint.java

```diff
@@ -147,4 +153,54 @@ public Response updateAndroidVariation(
 
         return Response.status(Status.NOT_FOUND).entity("Could not find requested Variant").build();
     }
+
+    /**
+     * Get the Android Variant for the given ID
+     *
+     * @param variantId id of {@link Variant}
+     * @return          requested {@link Variant}
+     * @statuscode 400 The requested Variant resource exists but it is not for Android
+     * @statuscode 404 The requested Android Variant resource does not exist
+     */
+    @GET
+    @Path("/{variantId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response findVariantById(@PathParam("variantId") String variantId) {
+        return doFindVariantById(variantId, AndroidVariant.class);
+    }
+
+    /**
+     * Delete the Android Variant
+     *
+     * @param variantId id of {@link Variant}
+     *
+     * @return no content or 404
+     *
+     * @statuscode 204 The Variant successfully deleted
+     * @statuscode 400 The requested Variant resource exists but it is not for Android
+     * @statuscode 404 The requested Android Variant resource does not exist
+     */
+    @DELETE
+    @Path("/{variantId}")
+    public Response deleteVariant(@PathParam("variantId") String variantId) {
+        return this.doDeleteVariant(variantId, AndroidVariant.class);
```

Comment:
I think using `this` here is confusing since the method is implemented in the superclass.